### DISCUSSION TITLE
Parametric sweep in transient simulation

### DISCUSSION THREAD

**Discussion by mikevacany** [ansys/pyaedt#4374] [Q&A] [answered]
Hello,

I cannot figure out how to make a parametric transient sweep from pyAEDT. It is seems to me it is not supported, can you confirm?

The issue is likely missing props["SweepDefinition"] for transient analysis and then fails `def _add_sweep(self, sweep_variable, equation, override_existing_sweep):` in `SolveSetup.py`

For example, I tried to define DC voltage variable ("vddqlp") to be swept:
```
cir["vddqlp"] = "0.6V"

# define voltage source
vdc = cir.modeler.components.components_catalog["Independent Sources:V_DC"].place("vdc")
vdc.parameters["DC"] = "vddqlp"
gnd1 = cir.modeler.components.create_gnd()
vdcp = cir.modeler.components.create_voltage_probe("vdcp")
vdc.pins[0].connect_to_component(gnd1.pins[0])
vdc.pins[1].connect_to_component(vdcp.pins[0])

trans_setup = cir.create_setup(setupname="TransientRun", setuptype="NexximTransient")
trans_setup.props["TransientData"] = ["0.01ns", "200ns"]

# try to add sweep variable <= likely not work!!!
trans_setup.add_sweep_count("vddqlp",0.4,0.8,10,"V")

cir.analyze_setup("TransientRun")
```

Thank you for advice.
Michael

**Comment by Samuelopez-ansys** [comment]
Hi @mikevacany ,

Not sure what you are trying to do.

A transient setup in Circuit can not have a parameter sweep variation. I think you want to run a Parametric analysis, then you need to create a parametric analysis like:

`setup1 = app.parametrics.add("w1", 0.1, 20, 0.2, "LinearStep")`

https://aedt.docs.pyansys.com/version/stable/examples/02-HFSS/Create_3d_Component_and_use_it.html#id1

**Comment by mikevacany** [comment]
Hi @Samuelopez-ansys,

I already tried what I suggest, but this does not lead to parameter sweep simulation (if I am not doing something fundamentally wrong).

My intention is to add sweep parameter into Transient simulation as I can do it in AEDT GUI, see screenshot below. If there is an alternative/recommended way how to reach the same, please advice.

![image](https://github.com/ansys/pyaedt/assets/163863445/268da54e-528b-4c62-87e3-91aeb6605043)

Michael

**Comment by Samuelopez-ansys** [comment, marked as answer]
@mikevacany You are right, this option is not available in PyAEDT, please add an issue if you need it. In the meantime, you can use the AEDT API:

```
app = Circuit(specified_version="2024.1")
setup = app.create_setup(setuptype="NexximTransient")
app.oanalysis.EditTransient(
    setup.name,
    [
        "NAME:SimSetup",
        "DataBlockID:=", setup.props["DataBlockID"],
        "OptionName:=", "(Default Options)",
        "AdditionalOptions:=", "",
        "AlterBlockName:=", "",
        "FilterText:=", "",
        "AnalysisEnabled:=", 1,
        "HasTDRComp:=", 0,
        [
            "NAME:OutputQuantities"
        ],
        [
            "NAME:NoiseOutputQuantities"
        ],
        "Name:=", setup.name,
        "TransientData:=", ["0.1ns", "10ns"],
        "TransientNoiseData:=", [False, "", "", 0, 1, 0, False, 1],
        "TransientOtherData:=", ["default", False],
        [
            "NAME:SweepDefinition",
            "Variable:=", "test",
            "Data:=", "LINC 5mV 10mV 10",
            "OffsetF1:=", False,
            "Synchronize:=", 0
        ]
    ])
```

**Reply by mikevacany** [reply]
Thanks @Samuelopez-ansys, this approach works for me.
